context/src/env/validator.rs

```diff
@@ -244,17 +253,17 @@ pub fn validate(ci_info: &CIInfo) -> EnvValidation {
     if let Some(branch_class) = &ci_info.branch_class {
         match (branch_class, ci_info.pr_number) {
             (BranchClass::PullRequest, None) => {
-                env_validation.add_issue(EnvValidationIssue::Invalid(
-                    EnvValidationIssueInvalid::CIInfoPRNumberMissing,
+                env_validation.add_issue(EnvValidationIssue::SubOptimal(
+                    EnvValidationIssueSubOptimal::CIInfoPRNumberMissing,
                 ));
             }
-            (BranchClass::Merge | BranchClass::ProtectedBranch, Some(..)) => {
-                env_validation.add_issue(EnvValidationIssue::Invalid(
-                    EnvValidationIssueInvalid::CIInfoPRNumberConflictsWithBranchClass,
+            (BranchClass::Merge | BranchClass::ProtectedBranch | BranchClass::None, Some(..)) => {
+                env_validation.add_issue(EnvValidationIssue::SubOptimal(
+                    EnvValidationIssueSubOptimal::CIInfoPRNumberConflictsWithBranchClass,
```

Comment:
I know you just moved this over. But I would expect this to actually be incorrect. Merge branch classes may have PR numbers. Depending on what setting you have configured, Trunk Merge will make a PR that for things that you enqueue and will merge those

I'm assuming we aren't using/surfacing these yet or we would be getting a bunch of suboptimal issues for this

Reply:
Good eye! looks like this was caught and resolved in https://github.com/trunk-io/analytics-cli/pull/199#discussion_r1876357400